### ISSUE TITLE
The constant in lower case is defined as a variable

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript-TmLanguage#542]
_From @dedmazayukr on November 19, 2017 21:20_

Why the constant in lower case is defined as a variable ? The constant must be constant, regardless of the register.

- VSCode Version: Version 1.18.1
- OS Version: Ubuntu 17.10

Steps to Reproduce:

1. Write for example: (lowercase and uppercase)
    `const variable = 'as variable.other.readwrite';`
     `const CONSTANT = 'as variable.other.constant';`

2. Press `ctrl+shift+p`, and select `Developer: Inspect TM Scope`

3. Click on word `variable` --> variable.other.readwrite (wrong)
![image](https://user-images.githubusercontent.com/22115961/32995597-8f387d7a-cd7f-11e7-81b7-2498691091c7.png)

4. Click on word `CONSTANT` --> variable.other.constant (correctly)
![image](https://user-images.githubusercontent.com/22115961/32995622-da314cb2-cd7f-11e7-8b72-41ccc073a317.png)

Reproduces without extensions: Yes
Languages: javascript and typescript

_Copied from original issue: Microsoft/vscode#38735_

**Comment by sheetalkamat** [comment]
Agreed, though this is not reported often and hence could stay as it is for now. Fixing this means copying multiple pattern rules and modifying just little to accommodate this. If there is enough demand this can be looked at later time.

**Comment by usernamehw** [comment]
Seriously?

**Comment by dedmazayukr** [comment]
What's going on here ?! How can you ignore the incorrect classification of variables and constants? I am shocked and disappointed! What are the arguments for closing this bug?

**Comment by sheetalkamat** [comment]
This issue is not being reported many times... Note that in declaration you can argue about const being const regardless of case etc, but in the end during its usage, these are the same heuristics we can apply given that its a regex grammar. If the issue gets reported multiple times and has lot of demand this can be done for the declaration site as that is the only feasible solution we have.. Given that we can not completely fix this issue and number of reports on this, this at moment would be as is..

**Comment by dedmazayukr** [comment]
This is done if peoples ask for a new function or feature. But this is a bug, I thought that the bug fixes are not dependent on the number of votes

**Comment by akamud** [comment]
This has been reported [at least 3 times](https://github.com/akamud/vscode-theme-onedark/issues?q=is%3Aissue+const+is%3Aclosed) for my theme. Please consider fixing this.

**Comment by akamud** [comment]
This has just been [reported again](https://github.com/akamud/vscode-theme-onedark/issues/86) for my theme.

**Comment by Budgiebrain994** [comment]
I'm also coming from @akamud 's theming, please address this issue

**Comment by sheetalkamat** [comment]
I still believe fixing this is just going to raise more coloring issues than not fixing. 

Eg. in code:
```typescript
const x = 10;
foo = x; // x cannot be tokenized as const here. That means the color here would be different from declaration. There is no way to do this using just grammar.
```

**Comment by akamud** [comment]
This would be one _less_ coloring issue. For comparison, Atom works like you described and it is better than not having `const` coloring at all.

**Comment by ShayanJavadi** [comment]
please fix this. literally the only thing keeping me from jumping ship from atom to vscode.
